falcon_policy/middleware.py

```diff
@@ -11,13 +11,18 @@ def __init__(self, config_dict):
 
     def process_resource(self, req, resp, resource, params):
         route = req.uri_template
-        roles_header = req.get_header('X-Roles', default='@unknown')
-        provided_roles = [role.strip() for role in roles_header.split(',')]
+
+        roles = req.context.get('roles')
```

Comment:
I am just thinking to myself if we could somehow sync this in a forward-compatible fashion to https://github.com/falconry/falcon/issues/1120. Maybe we could verify that _context_ is a ``dict``and assume class otherwise? With making it clear in the documentation what we do, and we don't of course :slightly_smiling_face: 

Or we could just clearly document that ``falcon-policy`` assumes that *req.context* is a dict for now.

Even the potential change in Falcon aside, clarifying or supporting the class case might be useful if custom middleware/authn/z machinery people are using is already dealing with a custom context class.